# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/mkinitcpio/hotfix.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- mkinitcpio-v39.2/Makefile.vanilla	2025-01-07 20:35:08.973342194 -0500
+++ mkinitcpio-v39.2/Makefile	2025-01-07 20:36:29.210009772 -0500
@@ -68,12 +68,12 @@
 	chmod 755 $(DESTDIR)/usr/bin/lsinitcpio $(DESTDIR)/usr/bin/mkinitcpio
 
 	install -m644 mkinitcpio.conf $(DESTDIR)/etc/mkinitcpio.conf
-	install -m755 -t $(DESTDIR)/usr/lib/initcpio init shutdown functions
-	install -m644 -t $(DESTDIR)/usr/lib/initcpio init_functions
+	install -m755 init shutdown functions -t $(DESTDIR)/usr/lib/initcpio
+	install -m644 init_functions -t $(DESTDIR)/usr/lib/initcpio
 	install -m644 udev/01-memdisk.rules $(DESTDIR)/usr/lib/initcpio/udev/01-memdisk.rules
 
 	cp -at $(DESTDIR)/usr/lib/initcpio hooks install
-	install -m644 -t $(DESTDIR)/usr/share/mkinitcpio mkinitcpio.d/*
+	install -m644 mkinitcpio.d/* -t $(DESTDIR)/usr/share/mkinitcpio
 
 	install -m644 systemd/mkinitcpio-generate-shutdown-ramfs.service \
 		$(DESTDIR)/usr/lib/systemd/system/mkinitcpio-generate-shutdown-ramfs.service
--- mkinitcpio-v39.2/Makefile.vanilla	2025-01-07 20:50:41.273352718 -0500
+++ mkinitcpio-v39.2/Makefile	2025-01-07 20:51:00.450019605 -0500
@@ -72,13 +72,13 @@
 	install -m644 init_functions -t $(DESTDIR)/usr/lib/initcpio
 	install -m644 udev/01-memdisk.rules $(DESTDIR)/usr/lib/initcpio/udev/01-memdisk.rules
 
-	cp -at $(DESTDIR)/usr/lib/initcpio hooks install
+	cp -a hooks install -t $(DESTDIR)/usr/lib/initcpio
 	install -m644 mkinitcpio.d/* -t $(DESTDIR)/usr/share/mkinitcpio
 
 	install -m644 systemd/mkinitcpio-generate-shutdown-ramfs.service \
 		$(DESTDIR)/usr/lib/systemd/system/mkinitcpio-generate-shutdown-ramfs.service
 	for target in $(addprefix $(DESTDIR),$(SHUTDOWN_DIRS)); do \
-		ln -s ../mkinitcpio-generate-shutdown-ramfs.service -t $$target || exit; \
+		ln -sf ../mkinitcpio-generate-shutdown-ramfs.service -t $$target || exit; \
 	done
 	install -m644 tmpfiles/mkinitcpio.conf $(DESTDIR)/usr/lib/tmpfiles.d/mkinitcpio.conf
 
@@ -87,7 +87,7 @@
 	install -m644 man/mkinitcpio.8 $(DESTDIR)/usr/share/man/man8/mkinitcpio.8
 	install -m644 man/mkinitcpio.conf.5 $(DESTDIR)/usr/share/man/man5/mkinitcpio.conf.5
 	install -m644 man/lsinitcpio.1 $(DESTDIR)/usr/share/man/man1/lsinitcpio.1
-	ln -s mkinitcpio $(DESTDIR)/usr/share/bash-completion/completions/lsinitcpio
+	ln -sf mkinitcpio $(DESTDIR)/usr/share/bash-completion/completions/lsinitcpio
 
 install-hooks: prepare
 	install -m644 libalpm/hooks/90-mkinitcpio-install.hook $(DESTDIR)/usr/share/libalpm/hooks/90-mkinitcpio-install.hook
--- mkinitcpio-v39.2/mkinitcpio.vanilla	2025-01-08 17:51:54.407252309 -0500
+++ mkinitcpio-v39.2/mkinitcpio	2025-01-08 17:51:58.683924615 -0500
@@ -33,9 +33,6 @@
 _optskiphooks=() _optaddhooks=() _hooks=() _optpreset=() _tmpfiles=() _generated=()
 declare -A _runhooks _addedmodules _modpaths _autodetect_cache
 
-# export a sane PATH
-export PATH='/usr/bin'
-
 # Sanitize environment further
 # GREP_OPTIONS="--color=always" will break everything
 # CDPATH can affect cd and pushd
